util/git/client.go

```diff
@@ -341,6 +341,19 @@ func (m *nativeGitClient) Init() error {
 		Name: git.DefaultRemoteName,
 		URLs: []string{m.repoURL},
 	})
+	if err != nil {
+		return err
+	}
+
+	// set automatic pack limit if specified
+	if countStr := os.Getenv(common.EnvGitAutopackLimit); countStr != "" {
```

Comment:
Just to know, what is the benefit to do it in that way?

Which approach  we should update a few functions signatures:

```golang
func (s *Service) checkoutRevision(gitClient git.Client, revision string, initConstants RepoServerInitConstants) (goio.Closer, error)

func checkoutRevision(gitClient git.Client, revision string, initConstants RepoServerInitConstants) error 
```
In the case of Client.Init() I have doubts, which signature you prefer?
- Init(autopackLimit int) or
- init(initConstants RepositoryInitConstants)
- another one 😅 

?
Thanks!

Reply:
> what is the benefit to do it in that way?

Clarity. We have a few env vars that are initialized deep in the code. It's easy to miss that they exist and that they impact functions, because they're not included in the signature of those functions. imo it's better to pass all the inputs explicitly and avoid implicit environment-based inputs whenever possible.

I think I prefer not modifying the `Init` signature but rather adding the limit as a field on `nativeGitClient` and then creating a new `ClientOpts` function to set the field.

```go
func WithEventHandlers(autopackLimit int) ClientOpts {
	return func(c *nativeGitClient) {
		c.autopackLimit = autopackLimit
	}
}
```

Reply:
I really like this approach!
Thanks a lot mate for the clarification and the help!!

Reply:
Done